goto-gcc -D_FORTIFY_SOURCE=3 : /usr/include/bits/stdio2.h:31:1: warning: function '__builtin_dynamic_object_size' is not declared

When compiling (sysstat) with `goto-gcc` with `-D_FORTIFY_SOURCE=3` there is warnings such as:
```
make: Entering directory '/usr/src/RPM/BUILD/sysstat-12.7.5'
goto-gcc -o sadf_misc.o -c -pipe -frecord-gcc-switches -Wall -fdiagnostics-color=always -fdiagnostics-color=always -g -O2 -flto=auto  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -Wall -Wstrict-prototypes -pipe -O2 -fno-tree-slp-vectorize -DSA_DIR=\"/var/log/sa\" -DSADC_PATH=\"/usr/lib64/sa/sadc\" -DHAVE_SENSORS   -DHAVE_SYS_SYSMACROS_H -DHAVE_LINUX_SCHED_H -DHAVE_SYS_PARAM_H -DUSE_NLS -DPACKAGE=\"sysstat\" -DLOCALEDIR=\"/usr/share/locale\" sadf_misc.c
/usr/include/bits/stdio2.h: In function 'sprintf':
/usr/include/bits/stdio2.h:31:1: warning: function '__builtin_dynamic_object_size' is not declared
                                  __glibc_objsize (__s), __fmt,
make: Leaving directory '/usr/src/RPM/BUILD/sysstat-12.7.5'
...
make: Entering directory '/usr/src/RPM/BUILD/sysstat-12.7.5'
goto-gcc -o sa_wrap.o -c -pipe -frecord-gcc-switches -Wall -fdiagnostics-color=always -fdiagnostics-color=always -g -O2 -flto=auto  -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=3 -Wall -Wstrict-prototypes -pipe -O2 -fno-tree-slp-vectorize -DSOURCE_SADC -DSA_DIR=\"/var/log/sa\" -DSADC_PATH=\"/usr/lib64/sa/sadc\" -DHAVE_SENSORS   -DHAVE_SYS_SYSMACROS_H -DHAVE_LINUX_SCHED_H -DHAVE_SYS_PARAM_H -DUSE_NLS -DPACKAGE=\"sysstat\" -DLOCALEDIR=\"/usr/share/locale\" sa_wrap.c
/usr/include/bits/strings_fortified.h: In function 'bcopy':
/usr/include/bits/strings_fortified.h:26:1: warning: function '__builtin_dynamic_object_size' is not declared
                                  __glibc_objsize0 (__dest));
make: Leaving directory '/usr/src/RPM/BUILD/sysstat-12.7.5'
```
While `-D_FORTIFY_SOURCE=2` does not exhibit the warnings.

CBMC version: 5.95.1
Operating system: ALT
GCC version: gcc version 13.2.1 20230817 (ALT Sisyphus 13.2.1-alt2) (GCC)

--- a/src/ansi-c/c_typecheck_expr.cpp
+++ b/src/ansi-c/c_typecheck_expr.cpp
@@ -3512,11 +3512,15 @@ exprt c_typecheck_baset::do_special_functions(
 
     return typecast_exprt(expr.arguments()[0], expr.type());
   }
-  else if(identifier=="__builtin_object_size")
+  else if(
+    identifier == "__builtin_object_size" ||
+    identifier == "__builtin_dynamic_object_size")
   {
-    // this is a gcc extension to provide information about
+    // These are gcc extensions to provide information about
     // object sizes at compile time
     // http://gcc.gnu.org/onlinedocs/gcc/Object-Size-Checking.html
+    // Our behavior is as if it was never possible to determine the object that
+    // the pointer pointed to.
 
     if(expr.arguments().size()!=2)
     {

--- a/src/ansi-c/compiler_headers/gcc_builtin_headers_mem_string.h
+++ b/src/ansi-c/compiler_headers/gcc_builtin_headers_mem_string.h
@@ -119,6 +119,7 @@ void* __builtin_memmove(void*, const void*, __CPROVER_size_t);
 void* __builtin_mempcpy(void*, const void*, __CPROVER_size_t);
 void* __builtin_memset(void*, int, __CPROVER_size_t);
 __CPROVER_size_t __builtin_object_size(const void*, int);
+__CPROVER_size_t __builtin_dynamic_object_size(const void*, int);
 int __builtin_popcount(unsigned);
 int __builtin_popcountimax(uintmax_t);
 int __builtin_popcountll(unsigned long long int x);

